### PR TITLE
ahhh

### PR DESCRIPTION
**Pull request by mush-zhang**
sql table leak 那个一直没弄好

### CODE REVIEW

File: src/storage/sql_table.cpp [hunk 1]
```diff
@@ -168,16 +172,96 @@ void SqlTable::Scan(const terrier::common::ManagedPointer<transaction::Transacti
...
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::BIGINT: {
+      auto val = type::TransientValuePeeker::PeekBigInt(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::DATE: {
+      // TODO(Schema-Change): find a way to handle it without codegen
```
Comment (rickyyx) [inline src/storage/sql_table.cpp:210]: why this one needs to be handled differently if it is a constant value? Or is it referring to the case when the default value is from some function like Now()?

File: src/storage/sql_table.cpp [hunk 2]
```diff
@@ -168,16 +172,96 @@ void SqlTable::Scan(const terrier::common::ManagedPointer<transaction::Transacti
...
+      break;
+    }
+    case type::TypeId::VARCHAR:
+    case type::TypeId::VARBINARY: {
+      auto val = terrier::type::TransientValuePeeker::PeekVarChar(transient_val);
+      memcpy(value_output, &val, val.size());
+      break;
+    }
+    default:
+      // TODO(Amadou): Add support for these types.
```
Comment (rickyyx) [inline src/storage/sql_table.cpp:237]: who is Amadou hah

File: src/storage/sql_table.cpp [hunk 3]
```diff
@@ -168,16 +172,96 @@ void SqlTable::Scan(const terrier::common::ManagedPointer<transaction::Transacti
...
+      auto julian_usec = util::TimeConvertor::ExtractJulianMicroseconds(val);
+      memcpy(value_output, &julian_usec, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::DECIMAL: {
+      auto val = type::TransientValuePeeker::PeekDecimal(transient_val);
+      memcpy(value_output, &val, type::TypeUtil::GetTypeSize(transient_val.Type()));
+      break;
+    }
+    case type::TypeId::VARCHAR:
```
Comment (rickyyx) [inline src/storage/sql_table.cpp:230]: what to do for varchar ?

**Review by rickyyx — APPROVED** [review]

File: src/storage/sql_table.cpp (hunk 1 above)
Comment (mush-zhang) [reply src/storage/sql_table.cpp:210]: you can specify a fixed date as default value . this is not for Now() which is a builtin

File: src/storage/sql_table.cpp (hunk 2 above)
Comment (mush-zhang) [reply src/storage/sql_table.cpp:237]: 做execution 那边的天才小哥 下学期去MIT读博了.... 上学期我学得要死要活的数学课他都考满分 :crying_cat_face: :sob:

File: src/storage/sql_table.cpp (hunk 3 above)
Comment (mush-zhang) [reply src/storage/sql_table.cpp:230]: fall through 跟varbinary一样

File: src/storage/sql_table.cpp (hunk 2 above)
Comment (mush-zhang) [reply src/storage/sql_table.cpp:237]: 这个是我从execution那边copy paste的

**Pull request merged**